src/dag/put.js

```diff
@@ -15,29 +14,35 @@ module.exports = (send) => {
 
   return promisify((dagNode, options, callback) => {
     if (typeof options === 'function') {
-      return setImmediate(() => callback(new Error('no options were passed')))
+      callback = options
+    } else if (options.cid && (options.format || options.hash)) {
+      return callback(new Error('Can\'t put dag node. Please provide either `cid` OR `format` and `hash` options.'))
+    } else if ((options.format && !options.hash) || (!options.format && options.hash)) {
+      return callback(new Error('Can\'t put dag node. Please provide `format` AND `hash` options.'))
```

Comment:
Turns out this is a breaking change as well (which is also one of the reasons CI fails at this time) Prior to this commit, it was okay to just have `format` with `hash` falling back to `sha2-256`.

Again, this would be a point of either introducing a breaking change and staying consistent with `js-ipfs` API, or leaving as is and live with the consequences of inconsistency. 

@alanshaw your call.